app/models/user.rb

```diff
@@ -1,2 +1,17 @@
 class User < ActiveRecord::Base
+  has_secure_password
+  has_many :questions, foreign_key: 'creator_id'
+  has_many :answers
+  validates :email, presence: true
+  validates :email, format: { with: /\A([^@\s]+)@((?:[-a-z0-9]+\.)+[a-z]{2,})\z/i, on: :create }
+
+  before_save :rename
+
+  def rename
+     self.name = email.slice(/^(.*?)@/).chop
```

Comment:
I find this a bit odd - it feels a bit strange to match the @ but then chop it. I might have done ```self.name = email.sub(/@.+/,'')```  - but maybe that's just personal preference.